src/dune/dune_file.ml

```diff
@@ -1989,9 +1990,22 @@ module Alias_conf = struct
   let decode =
     fields
       (let+ name = field "name" Alias.Name.decode
-       and+ loc = loc
        and+ package = field_o "package" Pkg.decode
-       and+ action = field_o "action" (located Action_dune_lang.decode)
+       and+ action =
+         field_o "action"
+           (let* syntax_version = Dune_lang.Syntax.get_exn Stanza.syntax
+            and+ loc = loc in
+            let deleted_in_version = (2, 0) in
+            if syntax_version < deleted_in_version then
+              located Action_dune_lang.decode
+            else
+              let what = "action field" in
+              let repl =
+                [ Pp.text "Use a rule stanza with the alias field instead" ]
+              in
+              Dune_lang.Syntax.Error.deleted_in loc Stanza.syntax
+                deleted_in_version ~what ~repl)
```

Comment:
Can't we use `Syntax.deleted_in` directly?

Reply:
The error will not look as good unless we can include `repl` in there. I guess can do that.

Reply:
Using it right now.

Reply:
I had a quick look and it seems that `extra_info` serves as `repl` for this function. Not sure about the naming `extra_info` though...